Automattic/jetpack
# Update my jetpack product statuses to be more feature based

## Proposed changes:

* This PR updates how some My Jetpack products perform plan checks to be more compatible with WordPress.com plan types - this is done by leveraging feature checks. Specifically, we are checking for features to determine if VideoPress, Social and AI are enabled in certain contexts.
* This PR hides the "Upgrade" button for the stats My Jetpack card on Atomic
* This PR removes the Creator card from My Jetpack on Atomic

## Jetpack product discussion


pfsHM7-ix-p2#comment-437

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






* Test this PR on a self-hosted Jetpack site, make sure the product active and inactive states still make sesne
* Test specifically VideoPress, AI and Social to make sure that free activation and purchases still work as expected and that the cards show as 'inactive' still when the feature/plan is absent.
* Test this on a WPCOM Atomic site with a Creator plan. You should have Backup, Protect, Social, VideoPress, Akismet, AI and Search all "Active" or "Needs Plugin" without any further action or purchase. Note that for Social and Protect to work properly, the site needs to be set to "Public" (see: p9o2xV-1r2-p2 for info on setting up a test Atomic site)
* The Atomic site should not show the Creator card
* The Atomic site should also not show an upgrade button for stats

![Screenshot 2024-02-29 at 12 09 18 PM](https://github.com/Automattic/jetpack/assets/18016357/5c8a9c38-e7a9-4773-b88e-65af90aa6ee6)
